### PR TITLE
Migrate from travis to github actions

### PR DESCRIPTION
**Pull request by peusebiu**
Use freeport package to get ports for testing zot servers

### CODE REVIEW

**Comment by peusebiu** [comment]
Do you want to run on more go versions?
Should I split this commit in two?

File: pkg/api/controller_test.go [hunk 1]
```diff
@@ -1677,46 +1740,56 @@ func TestHTTPReadOnly(t *testing.T) {
...
 				So(resp, ShouldNotBeNil)
 				So(resp.StatusCode(), ShouldEqual, 405)
 
 				//with invalid creds, it should fail
-				resp, _ = resty.R().SetBasicAuth("chuck", "chuck").Get(BaseURL4 + "/v2/")
+				resp, _ = resty.R().SetBasicAuth("chuck", "chuck").Get(baseURL + "/v2/")
 				So(resp, ShouldNotBeNil)
 				So(resp.StatusCode(), ShouldEqual, 401)
+
+				defer func(controller *api.Controller) {
```
Comment (andaaron) [inline pkg/api/controller_test.go:1777]: Is this change still needed after the update to use 'freeport'?

**Review by andaaron — APPROVED** [review]

File: pkg/api/controller_test.go (hunk 1 above)
Comment (peusebiu) [reply pkg/api/controller_test.go]: Yes, it works now. I have update it.

**Comment by rchamarthy** [comment]
> Do you want to run on more go versions?
> Should I split this commit in two? (one with the addition of freeport package and one with migration on github actions)

lets not worry about go versions. We can pick the latest and keep moving up.

**Review by andaaron — APPROVED** [review]

**Review by shimish2 — COMMENTED** [review]
Do you want to do same changes related to port on following test code:- 
pkg/extensions/search/cve/cve_test.go

**Comment by rchincha** [comment]
> 
> 
> Do you want to do same changes related to port on following test code:-
> pkg/extensions/search/cve/cve_test.go

wrt freeport?

**Comment by shimish2** [comment]
> > Do you want to do same changes related to port on following test code:-
> > pkg/extensions/search/cve/cve_test.go
> 
> wrt freeport?

Yes, freeport same as controller_test.go

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build container image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          tags: |
+            ghcr.io/${{ github.repository }}:latest
```
Comment (rchincha) [inline .github/workflows/ci-cd.yml:68]: Are we tagging each build with "latest" tag? This would be bad.

Comment (peusebiu) [reply .github/workflows/ci-cd.yml]: Ok, I removed it. Thank you.

Comment (peusebiu) [reply .github/workflows/ci-cd.yml]: I've put it back:D

**Comment by rchincha** [comment]
@peusebiu pls rebase and push again
